StripeConnect/StripeConnect/Source/Components/PaymentDetailsViewController.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  PaymentDetailsViewController.swift
+//  StripeConnect
+//
+//  Created by Mel Ludowise on 8/30/24.
+//
+
+import UIKit
+
+/**
+ Show details of a given payment and allow users to manage disputes and perform refunds.
+ */
+@_spi(DashboardOnly)
+public class PaymentDetailsViewController: UIViewController {
+    let webView: ConnectComponentWebView
+
+    public weak var delegate: PaymentDetailsViewControllerDelegate?
+
+    init(componentManager: EmbeddedComponentManager) {
+        webView = ConnectComponentWebView(
+            componentManager: componentManager,
+            componentType: .paymentDetails
+        )
+        super.init(nibName: nil, bundle: nil)
+        webView.addMessageHandler(OnLoadErrorMessageHandler { [weak self] value in
+            guard let self else { return }
+            self.delegate?.paymentDetailsLoadDidFail(self, withError: value.error.connectEmbedError)
+        })
+        // TODO: Add support for `setOnClose`
```

Comment:
Blocked on updating `OnSetterFunctionCalledMessageHandler`.

See https://github.com/stripe/stripe-ios/pull/4031#discussion_r1769634062

Reply:
This should no longer be blocked, but it can go in another PR.

Reply:
I think I'm going to delete the comment – I think we actually don't want to support `setOnClose` anyway since it should get handled in the app layer (we need to likely remove the X button from the webview)